#39-datepicker-and-textarea-AdminPage

AndrewPang321 opened the pull request:
- Implement DatePicker and TextArea using Angular Material in Admin Page
- Add 2 buttons for saving data
- Save relevant information into Firebase
- Chang to 40-space indentation
- Fixed indentation and redundant lines

Commit:
DatePicker&&TextArea

Implement DatePicker and TextArea

AndrewPang321 commented:
For clarification:
- In admin.html: added 2 datePickers(start & end date), 1 textArea(details) and 2 buttons(one for date and one for textArea).
- In admin.js: 2 functions(saveDate() & saveContent()) corresponding to 2 buttons in admin.html. Both save the information to the firebase under "events/eventName/admin/param" (for details, please take a look at 36_redesign_event_1 on FIrebase. Some tested data are displayed there for testing.)

chautsunman commented on app/js/admin.js:
```diff
@@ -48,17 +56,25 @@ angular.module("teamform-admin-app", ["firebase", "ngMaterial"])
+        //console.log($scope.startDate);
+        ref.update({'startDate': $scope.startDate.toString(), 'endDate': $scope.endDate.toString()});
+    };
+    $scope.saveContent = function() {
```
combine $scope.saveContent and $scope.saveDate function (1 save button for saving both the details and the dates)

chautsunman commented on app/js/admin.js:
```diff
@@ -7,8 +7,16 @@ $(document).ready(function() {
 .controller("AdminCtrl", function($scope, $firebaseObject, $firebaseArray) {
+
+    // Date
+    $scope.startDate = new Date();
```
display the saved dates if the event was created

chautsunman commented on app/js/admin.js:
```diff
@@ -7,8 +7,16 @@ $(document).ready(function() {
+    $scope.endDate = new Date();
+
+    // TextArea: details
+    $scope.details;
```
display the saved details if the event was created

chautsunman commented on app/admin.html:
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
+                        <label>Details: </label>
+                        <textarea ng-model="details" md-maxlength="150" rows="2" md-select-on-focus></textarea>
+                    </md-input-container>
+                    <button type="button" class="btn btn-primary" ng-click="saveContent()" >Save Details</button>
```
change `type="button"` into `type="submit"`

chautsunman commented on app/admin.html:
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
+                        <div layout-gt-xs="row">
+                            <div flex-gt-xs>
+                                <h4>Start Date: </h4>
+                                <md-datepicker ng-model="startDate" md-placeholder="Enter date"></md-datepicker>
```
change this field to be required using `ng-required="true"`

chautsunman commented on app/admin.html:
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
+
+                            <div flex-gt-xs>
+                                <h4>End Date: </h4>
+                                <md-datepicker ng-model="endDate" md-placeholder="Enter date"></md-datepicker>
```
change this field to be required using `ng-required="true"`

chautsunman commented on app/admin.html:
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
+                    <!-- TextArea -->
+                    <md-input-container class="md-block">
+                        <label>Details: </label>
+                        <textarea ng-model="details" md-maxlength="150" rows="2" md-select-on-focus></textarea>
```
change this field to be required using `ng-required="true"`

chautsunman commented:
display the saved details and dates in the event team page (eventteam.html)

Commit:
Fixed features

- Modify admin.js: combine two functions into one & use
$scope.startDate instead of $scope.startDate.toString() for displaying
the correct date format on eventteam.html
- modify admin.html & eventteam.js
- modify eventteam.html: display event’s information under the
EventName & using the correct date formate to display startDate &
endDate

AndrewPang321 commented:
Done.
- Modify admin.js: combine two functions into one & use
  $scope.startDate instead of $scope.startDate.toString() for displaying
  the correct date format on eventteam.html
- modify admin.html(change start/endDate and Details to ng-require="true") & eventteam.js(load info from firebase)
- modify eventteam.html: display event’s information under the
  EventName & using the correct date format with filtering to display startDate &
  endDate

Commit:
Minor Changes

- changes on DatePicker and TextArea
